File: src/CommandLineCommands/VersionCommand.cs

class VersionCommand : Command
{
    public VersionCommand()
    {
    }

    override public string GetCommandName()
    {
        return "version";
    }

    public override bool IsEmpty()
    {
        return false;
    }

    public List<Task<int>> ExecuteAsync(bool interactive)
    {
        Console.WriteLine($"Version: {VersionInfo.GetVersion()}");
        return new List<Task<int>> { Task.FromResult(0) };
    }
}

File: src/commonjs/CommandLineCommands/VersionCommand.ts

import { Command } from '../CommandLine/Command';
import { ConsoleHelpers } from '../Helpers/ConsoleHelpers';
import { VersionInfo } from '../Helpers/VersionInfo';

export class VersionCommand extends Command {
    public constructor() {
        super();
    }

    public GetCommandName(): string {
        return "version";
    }

    public IsEmpty(): boolean {
        return false;
    }

    public async ExecuteAsync(interactive: boolean): Promise<any> {
        return new Promise<number>((resolve) => {
            ConsoleHelpers.WriteLine(`Version: ${VersionInfo.GetVersion()}`, true);
            resolve(0);
        });
    }
}